GitEnlistmentManager/DTOs/Commands/RefreshTreeviewCommand.cs

```diff
@@ -0,0 +1,28 @@
+﻿using GitEnlistmentManager.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+
+namespace GitEnlistmentManager.DTOs.Commands
+{
+    internal class RefreshTreeviewCommand : ICommand
+    {
+        public bool OpenNewWindow { get; set; } = false;
+
+        public void ParseArgs(GemNodeContext nodeContext, Stack<string> arguments)
+        {
+        }
+
+        public async Task<bool> Execute(GemNodeContext nodeContext, MainWindow mainWindow)
+        {
+            if (mainWindow.gem.ReloadSettings())

```

Comment:
fix: Instead of using mainWindow.gem here, can you use nodeContext.RepoCollection.Gem.ReloadSettings?
That way you don't need to make the gem in mainWindow public.